docs/HowTo/write/hints.md

```diff
@@ -0,0 +1,54 @@
+---
+description: Using hints for off-circuit computations
+---
+
+# Compiler hints
+
+Instead of computing some value within a circuit, it is sometimes more optimal
+to compute the value off-circuit and only verify the correctness of the
+computation in a circuit. `gnark` provides a way to provide such verifiable
+"advice wires" in the way of *hints*. From the prover point of view, these are
+essentially input variables which are provided by hint function instead of the
+user.
```

Comment:
```suggestion
Instead of computing some value within a circuit, it's sometimes more optimal
to compute the value off-circuit and only verify the correctness of the
computation in a circuit. `gnark` allows you to do this through *hints*. From the prover's point of view, hints are
essentially input variables provided by a hint function instead of the
user.
```